staging/src/k8s.io/kubectl/pkg/cmd/clusterinfo/clusterinfo_dump.go

```diff
@@ -170,7 +170,17 @@ func (o *ClusterInfoDumpOptions) Run() error {
 		return err
 	}
 
-	if err := o.PrintObj(nodes, setupOutputWriter(o.OutputDir, o.Out, "nodes.json")); err != nil {
+	var fileExtension string
+	switch o.PrintFlags.UsedOutputFormat() {
```

Comment:
Use `o.PrintFlags.OutputFormat` to get what you want.

Reply:
Ok